### PR TITLE
Improvements on Authoring Docks UX

### PR DESCRIPTION
**Pull request by ederign**
Depends on:
https://github.com/uberfire/uberfire/pull/155
https://github.com/uberfire/uberfire-extensions/pull/59

### CODE REVIEW

File: kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        DataModelerContext context = dataModelerWBContext.getActiveContext();
+        if (shouldDisplayWestDocks(context)) {
+            uberfireDocks.enable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+        } else {
+            uberfireDocks.disable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+        }
+    }
+
+    private boolean shouldDisplayWestDocks(DataModelerContext context) {
+        return context != null && context.getEditionMode() == DataModelerContext.EditionMode.GRAPHICAL_MODE;
```
Comment (ederign) [inline kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java]: @krisv as I told you that was my first time with Authoring Perspective, so I need help! :)

That rule comes from here https://github.com/droolsjbpm/kie-wb-common/blob/3e70fc6d8df3f9c589c54e6d2e52c8267d69b783/kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/common/domain/BaseDomainEditor.java#L123-L142

is that correct?

File: kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    private String authoringPerspectiveIdentifier;
+
+    private UberfireDock projectExplorerDock;
+
+
+    public void perspectiveChangeEvent(@Observes UberfireDockReadyEvent dockReadyEvent) {
+        if (authoringPerspectiveIdentifier != null && dockReadyEvent.getCurrentPerspective().equals(authoringPerspectiveIdentifier)) {
+            if (projectExplorerDock != null) {
+                uberfireDocks.expand(projectExplorerDock);
```
Comment (ederign) [inline kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java]: This was necessary to open(expand) the project explorer dock every time that we open a perspective.

The authoring doesn't work without that -> (http://prntscr.com/862cnq)

File: kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringNoContextNavigationPerspective.java [hunk 3]
```diff
@@ -77,37 +81,38 @@
...
+                "true");
+        return placeRequest;
+    }
 
-        perspective.getRoot().insertChild( CompassPosition.WEST,
-                                           west );
+    @Perspective
+    public PerspectiveDefinition getPerspective() {
+        final PerspectiveDefinitionImpl perspective = new PerspectiveDefinitionImpl( MultiListWorkbenchPanelPresenter.class.getName() );
+        perspective.setName( constants.project_authoring() );
```
Comment (ederign) [inline kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringNoContextNavigationPerspective.java]: We've removed project explorer ("org.kie.guvnor.explorer")  and add it in the west dock.

**Comment by manstis** [comment]
Don't forget license header - @mbiarnes spent a long time adding them a while ago...

**Comment by ederign** [comment]
@manstis ok! https://www.youtube.com/watch?v=2nmwwKCGQus

**Comment by porcelli** [comment]
http://github.com/uberfire/uberfire-extensions/commit/3564cf9ad

**Comment by porcelli** [comment]
http://github.com/droolsjbpm/kie-wb-distributions/commit/f4f92f10e

**Pull request closed without merging**